src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/validate.ts

```diff
@@ -20,35 +24,42 @@ export const validate = (
 ): ESQLMessage[] => {
   const messages: ESQLMessage[] = [];
 
-  // Run standard argument validation
-  messages.push(...validateCommandArguments(command, ast, context, callbacks));
-
-  // Cast to RERANK command for type-specific validation
-  const rerankCommand = command as ESQLAstRerankCommand;
+  const { query, targetField, location, inferenceId } = command as ESQLAstRerankCommand;
+  const rerankExpressionType = getExpressionType(
+    query,
+    context?.fields,
+    context?.userDefinedColumns
+  );
 
-  // Validate that query text is provided
-  if (!rerankCommand.query) {
+  // check for supported query types
+  if (!supportedQueryTypes.includes(rerankExpressionType)) {
     messages.push({
-      location: command.location,
-      text: i18n.translate('kbn-esql-ast.esql.validation.rerankMissingQuery', {
-        defaultMessage: '[RERANK] Query text is required.',
+      location: 'location' in query ? query?.location : location,
+      text: i18n.translate('kbn-esql-ast.esql.validation.rerankUnsupportedFieldType', {
+        defaultMessage: '[RERANK] query must be of type [text]. Found [{rerankExpressionType}]',
```

Comment:
```suggestion
        defaultMessage: 'RERANK query must be of type text. Found {rerankExpressionType}',
```

No square brackets 😄